### PR TITLE
Fix logo on pdf for RTL

### LINKED ISSUE #27265: The Logo of PDFs are NOK with RTL lanaguage
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

When downloading PDFs with RTL language => NOK
This issue occurs with an invoice & Delivery slip & the credit slip

**Credit slip**
image
**Delivery slip**
image
**invoice**
image

In the PS1778 => ok
image

### Expected behavior

_No response_

### Steps to reproduce

1. Go to BO > Your profile & select the Arabic language
2. Go to BO > Order > Generate an invoice & download it
3. See an error
4. Generate a delivery slip & download it
5. See an error
6. Generate a credit slip & download it
7. See an error

### PrestaShop version(s) where the bug happened

1.7.8.0, 1.7.8.2, 1.7.8.3build1

### PHP version(s) where the bug happened

7.2

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by MeKeyCool**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | In PDF export with RTL language, logo was broken.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #27265 and probably #27794
| Related PRs       | https://github.com/PrestaShop/TCPDF/pull/1; https://github.com/tecnickcom/TCPDF/pull/510
| How to test?      | Cf. detailed description
| Possible impacts? | PDF exports

# How to test

## Environment setup

From a fresh install, to test :

- "**Generate an invoice**" pass some orders to "Payment accepted" status.
- "**Generate a credit slip**", cf. user documentation : https://doc.prestashop.com/display/PS17/Credit+Slips
- "**Generate a delivery slip**", pass some orders to "Shipped" status.

## Steps to reproduce

- Go to BO > Your profile & select the Arabic language
- Go to BO > Order > Generate an invoice & download it
- See logo
- Generate a delivery slip & download it
- See logo
- Generate a credit slip & download it
- See logo

:question: @PrestaShop/qa-team , sorry I found #27794 after my tests and didn't check on my local install. What is the process if a PR should fix two issues ? May I reproduce step to reproduce of both in this description ?

### CODE REVIEW

**Comment by MeKeyCool** [comment]
@atomiix , do you think this fork may solve this issue as well : https://github.com/PrestaShop/PrestaShop/issues/27794 ?

**Comment by atomiix** [comment]
@MeKeyCool yes, it should!

**Review by atomiix — DISMISSED** [review]

File: composer.json
Comment (NeOMakinG) [inline composer.json]: Are you sure about this? Shouldn't we wait for a proper release?

Comment (MeKeyCool) [reply composer.json]: Yup, it is the branch fixed on prestashop fork : https://github.com/PrestaShop/TCPDF/pull/1

I discussed with @atomiix about a dedicated branch to avoid conflicts but it doesn't look necessary. 
Maybe cause the fix should be applied soon to the library :thinking:  (https://github.com/tecnickcom/TCPDF/pull/510)

Comment (mparvazi) [reply composer.json]: It's like a dead project!
Really boring to contribute on tcpdf.

Comment (MeKeyCool) [reply composer.json]: Hmmm ... Sorry but I don't understand the point.

Do you ask to create another branch ? A tag ? Should we just wait and let the bug ?

Comment (mparvazi) [reply composer.json]: I think it's better to wait.
Idk technically is it normal to use a fork branch of tcpdf on PS or not?
Because if we had, you'll have my guarantee to fix all RTL problems just in a second :)

Comment (MeKeyCool) [reply composer.json]: @PrestaShop/prestashop-maintainers and especially @eternoendless , do you have an opinion about this ?

I think, in some cases we won't be able to just wait but Idon't know about the conditions/limit forking our dependencies. I don't think it would be a good idea to allow our forks diverging from original projects ^^' ...

Comment (eternoendless) [reply composer.json]: It looks like the patch was made at the tip of TCPDF's main branch, which at the time of writing is 6.4.4 (https://github.com/tecnickcom/TCPDF/commit/42cd0f9786af7e5db4fcedaa66f717b0d0032320). Shouldn't the patch have been made on top of 6.2.12 to minimize the risk of regressions?

In any case, a branch named `6.4.4-patch` or `6.2.12-patch` should be created in the temporary fork, instead of `main`.

@mparvazi we use temporary forks to provide important bug fixes when the original library is slow to accept them. Once the fix is included in an official release, we will revert back to the original library and archive the fork. In general, it's not a good idea to fork away and continue development, because if you do you will have to take over the maintenance charge of the library, and we don't want to do that.

Comment (atomiix) [reply composer.json]: @eternoendless we're already using 6.4.4 in 8.0.x and 6.3.5 in 1.7.8.x, I think it's okay to update to 6.4.4.

It's only a naming convention for the branch name, right? I created the branch 6.4.4-patch anyway.

Comment (eternoendless) [reply composer.json]: @atomiix 

> we're already using 6.4.4 in 8.0.x

Yes but this PR is targeting 1.7.8.x

> It's only a naming convention for the branch name, right?

Not just. If we ever need to supply two different patches (one for 1.7.8.x and another for 8.x based on different versions), we'll need two different branches. Not having the main branch diverge from the original repo will also help if we need to sync forks, etc.

**Review by atomiix — APPROVED** [review]

**Review by mparvazi — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @MeKeyCool ,

Tested #27265 and #27794 on 1.7.8.x.
Tested with Persian and Arabic languages.
Tested with multiple sizes of logo and types (png, jpg).
Tested again and switched with non-RTL languages, it is looking good too.

Logo is well displayed in Credit Slips:
Screenshot 2022-08-09 at 15 26 53

in Invoices:
Screenshot 2022-08-09 at 15 36 46

and Delivery Slips:
Screenshot 2022-08-09 at 15 27 18

Product images are well displayed in Delivery Slips:
Screenshot 2022-08-09 at 15 32 27

It is QA ✅ 
Thanks!

**Review by jolelievre — APPROVED** [review]

**Comment by jolelievre** [comment]
Thanks @florine2623 and @MeKeyCool

**Comment by leemyongpakvn** [comment]
@MeKeyCool [tecnickcom/TCPDF](https://github.com/tecnickcom/TCPDF/commit/cc54c1503685e618b23922f53635f46e87653662) was bumped to 6.5.0 on Aug 12

**Pull request merged**
